backend/WebApi/Helpers/InputValidator.cs

```diff
@@ -0,0 +1,49 @@
+﻿using Domain.Entities;
+using System;
+using System.Collections.Generic;
+using System.Linq;
+using System.Threading.Tasks;
+
+namespace WebApi.Helpers
+{
+    public class InputValidator
+    {
+        public static bool ValidateRestaurantSortOrder(string sortOrder, Coords userCoordinates = null)
+        {
+            if (userCoordinates == null && (sortOrder == "dist" || sortOrder == "dist_desc"))
+            {
+                throw new System.ArgumentNullException("Invalid user coordinates.");
+            }
+
+            return sortOrder switch
+            {
+                "dist" => true,
+                "dist_desc" => true,
+                "name" => true,
+                "name_desc" => true,
+                null => true,
+                _ => throw new System.ArgumentException("Invalid sort order"),
+            };
```

Comment:
We could store the valid sorting types in an enum or an array, and then check if the type passed is contained there, for example using [this method](https://docs.microsoft.com/en-us/dotnet/api/system.enum.isdefined?view=net-5.0).
Why? If we want to add a new value to sort, we should change the validator class, although the logic has not changed. This way we just change the correct file just to update the list of correct inputs. It is not necessary to do it but just as a comment. Very good job!!

Reply:
I approve the changes, but I just left the comment!

Reply:
Made it use an Enum